src/clj/muuntaja/core.clj

```diff
@@ -82,8 +82,9 @@
                 ;; optional guard on empty imput
                 (if (and allow-empty-input-on-decode? decode?)
                   (fn [^InputStream is charset]
-                    (if (pos? (.available is)) ($ is charset)))
-                  $))
+                    (if (and (not (nil? is)) (pos? (.available is))) ($ is charset)))
+                  (fn [^InputStream is charset]
+                    (if-not (nil? is) ($ is charset)))))
```

Comment:
Now `:allow-empty-input-on-decode? false` doesn't do anything, or does it? When the server passes in `:body nil`, this happily returns `nil`, so empty inputs are allowed anyway.

That said, I'm not sure if there needs to be an option for disabling empty request bodies.